transformer/dockerfilegenerator/dotnetcoredockerfilegenerator.go

```diff
@@ -0,0 +1,492 @@
+		}
+		ir := irtypes.IR{}
+		irPresent := true
+		err = a.GetConfig(irtypes.IRConfigType, &ir)
+		if err != nil {
+			irPresent = false
+			logrus.Debugf("unable to load config for Transformer into %T : %s", ir, err)
+		}
+		ports := ir.GetAllServicePorts()
+		var dotNetCoreTemplateConfig DotNetCoreTemplateConfig
+		dotNetCoreTemplateConfig.AppName = a.Name
+		var csprojRelFilePath string
+		if len(a.Paths[DotNetCoreCsprojFilesPathType]) == 1 {
+			csprojRelFilePath, err = filepath.Rel(a.Paths[artifacts.ProjectPathPathType][0], a.Paths[DotNetCoreCsprojFilesPathType][0])
+			if err != nil {
+				logrus.Errorf("Unable to convert csproj file path %s to be relative : %s", a.Paths[DotNetCoreCsprojFilesPathType][0], err)
+				continue
+			}
+		} else {
+			csprojRelFilePath = getCsprojFileForService(a.Paths[DotNetCoreCsprojFilesPathType], a.Paths[artifacts.ProjectPathPathType][0], a.Name)
+		}
+		publishProfileFilesPath := findPublishProfiles(filepath.Join(a.Paths[artifacts.ProjectPathPathType][0], csprojRelFilePath))
+
+		publishProfileFilePath := getPublishProfile(publishProfileFilesPath, a.Name, a.Paths[artifacts.ProjectPathPathType][0])
+		if publishProfileFilePath != "" {
+			if publishUrl, err := parsePublishProfileFile(filepath.Join(a.Paths[artifacts.ProjectPathPathType][0], publishProfileFilePath)); err == nil {
+				dotNetCoreTemplateConfig.PublishProfileFilePath = publishProfileFilePath
+				dotNetCoreTemplateConfig.PublishUrl = publishUrl
+			}
+		}
+		dotNetCoreTemplateConfig.CsprojFilePath = csprojRelFilePath
+		projectName := strings.TrimSuffix(filepath.Base(dotNetCoreTemplateConfig.CsprojFilePath), filepath.Ext(dotNetCoreTemplateConfig.CsprojFilePath))
+		jsonFiles, err := common.GetFilesByExt(a.Paths[artifacts.ProjectPathPathType][0], []string{".json"})
+		if err != nil {
+			logrus.Debugf("Error while finding json files %s", err)
+		}
+		for _, jsonFile := range jsonFiles {
+			if filepath.Base(jsonFile) == launchSettingsJSON {
+				launchSettings := LaunchSettings{}
+				if err := common.ReadJSON(jsonFile, &launchSettings); err != nil {
+					logrus.Errorf("Unable to read the launchSettings.json file: %s", err)
+					continue
+				}
+				if launchSettings.Profiles[projectName].ApplicationURL != "" {
+					Urls := strings.Split(launchSettings.Profiles[projectName].ApplicationURL, ";")
+					for _, url := range Urls {
+						portStr := portRegex.FindAllString(url, 1)[0]
+						port, err := strconv.ParseInt(portStr, 10, 32)
+						if err != nil {
+							logrus.Errorf("Error while converting the port from string to int : %s", err)
+						} else if strings.HasPrefix(url, "https://") {
+							dotNetCoreTemplateConfig.HTTPSPort = int32(port)
+						} else if strings.HasPrefix(url, "http://") {
+							dotNetCoreTemplateConfig.HTTPPort = int32(port)
+						}
+					}
+					ports = append(ports, dotNetCoreTemplateConfig.HTTPPort)
+				}
+			}
+			if filepath.Base(jsonFile) == packageJSONFile {
```

Comment:
Is it okay if the package json is in any folder? doesn't it need to be in the folder where the startup csproj file is present?

Reply:
In [some cases](https://github.com/konveyor/move2kube-demos/tree/main/samples/dotnet5/dotnet5angular/ClientApp), the package.json is present inside the `ClientApp` folder inside the folder where the csproj file is present.